ios/engine/KMEI/KeymanEngine/Classes/Keyboard/KeymanWebViewController.swift

```diff
@@ -217,18 +217,24 @@ extension KeymanWebViewController {
 
   func setText(_ text: String?) {
     var text = text ?? ""
+
     // Remove any system-added LTR/RTL marks.
     text = text.replacingOccurrences(of: "\u{200e}", with: "") // Unicode's LTR codepoint
     text = text.replacingOccurrences(of: "\u{200f}", with: "") // Unicode's RTL codepoint (v1)
     text = text.replacingOccurrences(of: "\u{202e}", with: "") // Unicode's RTL codepoint (v2)
 
-    // JavaScript escape-sequence encodings.
-    text = text.replacingOccurrences(of: "\\", with: "\\\\")
-    text = text.replacingOccurrences(of: "'", with: "\\'")
-    text = text.replacingOccurrences(of: "\n", with: "\\n")
+    do {
+      let encodingArray = [ text ];
+      let jsonString = try String(data: JSONSerialization.data(withJSONObject: encodingArray), encoding: .utf8)!
+      let start = jsonString.index(jsonString.startIndex, offsetBy: 2)
```

Comment:
Why embedded in an array?

Reply:
This mirrors other functions requiring similarly-encoded Strings elsewhere in the class.  You can't pass in a raw string for this approach - it needs to be either an array or a dictionary, and other functions opted for the 'encodingArray' approach seen above.